### PR TITLE
Check for job.Schedule.DayOfMonth nullability

### PR DESCRIPTION
**Pull request by tkant**
Then mask ? else use job.Schedule.Month.Day
Added unit-test case

### CODE REVIEW

**Comment by tkant** [comment]
@ProTip Please review, this fixes: https://github.com/terraform-providers/terraform-provider-rundeck/pull/34

**Comment by tkant** [comment]
@ProTip Who else can I reach out??

File: rundeck/resource_job.go [hunk 1]
```diff
@@ -801,12 +801,16 @@ func jobToResourceData(job *JobDetail, d *schema.ResourceData) error {
...
+			schedule = append(schedule, job.Schedule.Month.Day)
+		} else {
+			schedule = append(schedule, "?")
+		}
 		schedule = append(schedule, job.Schedule.Month.Month)
 		if job.Schedule.WeekDay != nil {
 			schedule = append(schedule, job.Schedule.WeekDay.Day)
 		} else {
-			schedule = append(schedule, "*")
+			schedule = append(schedule, "?")
```
[the remark this answers is not in the document]
Comment (tkant) [reply rundeck/resource_job.go:813]: Sure @ProTip, lemme check.

[the remark this answers is not in the document]
Comment (tkant) [reply rundeck/resource_job.go]: @ProTip Updated as you suggested. 
Thanks!

**Comment by tkant** [comment]
Updated PR as suggested

**Comment by tkant** [comment]
@ProTip Can you please check this again.

**Comment by fdevans** [comment]
If you can resolve the conflicts on resource_job.go I can test.  Also please include any updates to documentation and steps to validate it's working.

**Comment by fdevans** [comment]
@tkant Checking in again to see if you can add documentation and validation tests.

**Comment by fdevans** [comment]
This was refactored in a more recent revamp.  Let us know if there are still issues.

**Pull request closed without merging**